src/traces/pie/attributes.js

```diff
@@ -180,6 +180,33 @@ module.exports = {
         description: 'Sets the font used for `textinfo` lying outside the pie.'
     }),
 
+    title: {
+        valType: 'string',
+        dflt: '',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the title of the pie chart.',
+            'If it is empty, no title is displayed.'
+        ].join(' ')
+    },
+    titleposition: {
+        valType: 'enumerated',
+        values: ['inhole', 'outside'],
+        dflt: 'outside',
```

Comment:
Can we make this look a bit more like [`textposition`](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/attributes.js#L512)? I'm not sure *quite* all of those options make sense, but most of them do. What you have now corresponds to `'top center'` and `'middle center'`, and it's fine to start with just those two options, but I can certainly see wanting `'top left'`, `'top right'`, and `'bottom (left|center|right)'` in the future.

Also I'd remove the `dflt` here and make it dependent on `hole`: if it's zero, default to `'top center'`, otherwise `'middle center'`.

Reply:
> What you have now corresponds to 'top center' and 'middle center', and it's fine to start with just those two options, but I can certainly see wanting 'top left', 'top right', and 'bottom (left|center|right)' in the future.

> Also I'd remove the dflt here and make it dependent on hole: if it's zero, default to 'top center', otherwise 'middle center'.

voting :+1:

Reply:
Done, added all options mentioned above.